textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/RegexPredicate.java

```diff
@@ -1,88 +1,58 @@
-/**
- * 
- */
 package edu.uci.ics.textdb.dataflow.common;
 
-import java.util.ArrayList;
 import java.util.List;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apache.lucene.analysis.Analyzer;
 
-import edu.uci.ics.textdb.api.common.IField;
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.FieldType;
 import edu.uci.ics.textdb.api.common.IPredicate;
-import edu.uci.ics.textdb.api.common.ITuple;
-import edu.uci.ics.textdb.common.field.Span;
-import edu.uci.ics.textdb.common.field.StringField;
-import edu.uci.ics.textdb.common.field.TextField;
+import edu.uci.ics.textdb.api.storage.IDataStore;
 
 /**
+ * This class is responsible for calculating regex match. <br>
+ * It uses RegexToGramQueryTranslator to translate a regex to a query in lucene. <br>
+ * 
+ * @author Zuozhi Wang
+ * @author Shuying Lai
  * @author sandeepreddy602
  *
  */
-public class RegexPredicate implements IPredicate{
-
-    private final String fieldName;
-    
-    private final String regex;
-    private final Pattern pattern;
-    private Matcher matcher;
+public class RegexPredicate implements IPredicate {
 
-	public RegexPredicate(String regex, String fieldName){
-        this.regex = regex;
-        this.fieldName = fieldName;
-        pattern = Pattern.compile(regex);
-    }
+	private String regex;
+	private List<String> fieldNameList;
 	
-    public String getFieldName() {
-		return fieldName;
+	private Analyzer analyzer;

```

Comment:
We decided to use `luceneAnalyzer' as a variable name instead of `analyzer` to make the Lucene dependency more clear.